### PR TITLE
Sc 106980/offer card

### PR DESCRIPTION
**Pull request by michael-velsigne-QTW**
## Description

Create Offer Card component with basic layout structure - including heading, location, star rating & tags default
With optional image gallery and offers

## Related issues

[SC 106980](https://app.shortcut.com/qantasaccommodation/story/106980)

### Default view:
image

### With all view:
image

### Mobile view:
image

### CODE REVIEW

File: src/components/OfferCard/OfferCard.test.jsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    render(
+      <OfferCard {...props} starRatingProps={overriddenStarRatingProps} />,
+    );
+    const stars = screen.getAllByTitle(ratingTypeIcon);
+
+    expect(stars.length).toEqual(5);
+  },
+);
+
+// Todo as not getByText
```
Comment (ajlu) [inline src/components/OfferCard/OfferCard.test.jsx:90]:
```suggestion
```

File: src/components/OfferCard/components/InclusionsGroup/InclusionsGroup.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Box, FeatureListItem, NakedButton, Text } from 'components';
+
+const InclusionGroup = ({
+  inclusionsLabel,
+  inclusionsCallToActionLabel,
+  onClickInclusionsCallToAction,
+  ...inclusionsProps
```
Comment (ajlu) [inline src/components/OfferCard/components/InclusionsGroup/InclusionsGroup.jsx:9]:
```suggestion
  inclusions
```
Can destructure `inclusions` prop as well?

File: src/components/OfferCard/components/InclusionsGroup/InclusionsGroup.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+}) => {
+  return (
+    <>
+      <Box mb={3}>
+        <Text fontSize="base">{inclusionsLabel}</Text>
```
Comment (ajlu) [inline src/components/OfferCard/components/InclusionsGroup/InclusionsGroup.jsx:14]:
```suggestion
        <Text>{inclusionsLabel}</Text>
```
`base` should already be the default font size?

File: src/components/OfferCard/components/InclusionsGroup/InclusionsGroup.jsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      </Box>
+
+      {inclusionsProps.inclusions.map((item, key) => (
+        <FeatureListItem key={key} {...item} />
+      ))}
+
```
Comment (ajlu) [inline src/components/OfferCard/components/InclusionsGroup/InclusionsGroup.jsx:20]:
```suggestion
      {inclusions.map((item, key) => (
        <FeatureListItem key={key} {...item} />
      ))}
```
remove extra lines

File: src/components/Tags/components/Tag/Tag.jsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import styled from '@emotion/styled';
+import { color, fontSize } from 'styled-system';
+
+const ListItem = styled.li`
+  display: inline;
+  ${color}
+  ${fontSize}
+`;
+
+const Tag = (props) => (
```
Comment (ajlu) [inline src/components/Tags/components/Tag/Tag.jsx:12]: Noticed this component is nested under `src/components/Tags` but would still be exported from roo-ui.

In this case, can `Tag` be moved to the same level as `Tags` - check the existing `List / ListItem` components
And similarly, please add a storybook and README for `Tag`

Same for `InclusionGroup` as well if it's intended to be exported from roo-ui?

File: src/components/OfferCard/components/InclusionsGroup/InclusionsGroup.jsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      )}
+    </>
+  );
+};
+
+InclusionGroup.propTypes = {
+  inclusions: PropTypes.arrayOf(PropTypes.shape(FeatureListItem.propTypes)),
+  inclusionsLabel: PropTypes.string.isRequired,
+  inclusionsCallToActionLabel: PropTypes.string,
+  onClickInclusionsCallToAction: PropTypes.func,
```
Comment (ajlu) [inline src/components/OfferCard/components/InclusionsGroup/InclusionsGroup.jsx:39]:
```suggestion
  label: PropTypes.string.isRequired,
  callToAction: PropTypes.shape({
    label: PropTypes.string,
    onClick: PropTypes.func,
  }),
```
potential props naming change? Given the component is called `InclusionsGroup`, its known the props are related to inclusions

File: src/components/OfferCard/OfferCard.test.jsx [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+    <div>
+      <img alt="Offer 1" />
+      <img alt="Offer 2" />
+      <img alt="Offer 3" />
+      <img alt="Offer 4" />
+    </div>
+  ),
+};
+
+test('renders heading', () => {
```
Comment (ajlu) [inline src/components/OfferCard/OfferCard.test.jsx:59]: Add accessibility testing for `OfferCard` and `InclusionsGroup` as well?

File: src/components/OfferCard/OfferCard.jsx [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+};
+
+const OfferCard = (props) => {
+  return (
+    <>
+      <Box borderRadius="10px" backgroundColor="white">
+        <Flex flexDirection={['column-reverse', null, 'row']}>
+          <Box display="flex" flex="1" p={6} pb={[6, null, 0]}>
+            <Flex
+              flexDirection={['column', null, 'column']}
```
Comment (ajlu) [inline src/components/OfferCard/OfferCard.jsx:24]:
```suggestion
              flexDirection="column"
```
If the styling is the same for all screens sizes, don't need a responsive array? applies for other similar stylings elsewhere

File: src/components/Tags/Tags.test.jsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+import Tags from './';
+
+test('renders Tags with children and correctly', () => {
+  render(
+    <Tags>
+      <Tags.Tag>Hello</Tags.Tag>
+      <Tags.Tag>World</Tags.Tag>
+    </Tags>,
+  );
+  const listItems = screen.getAllByRole('listitem');
```
Comment (ajlu) [inline src/components/Tags/Tags.test.jsx:12]:
```suggestion

  expect(screen.getByRole('list')).toBeInTheDocument()
  const listItems = screen.getAllByRole('listitem');
```
Use `list` to test for `<Tags />`

File: src/components/OfferCard/components/InclusionsGroup/InclusionsGroup.jsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Box, FeatureListItem, NakedButton, Text } from 'components';
+
+const InclusionGroup = ({
```
Comment (ajlu) [inline src/components/OfferCard/components/InclusionsGroup/InclusionsGroup.jsx:5]: roo-ui being a component library should remain generic without any business/product knowledge, so thinking this should be renamed to align with `FeatureListItem`?

`FeatureListItem` is mainly used for inclusions but it was introduced in roo-ui with a generic name

```suggestion
const FeatureListGroup = ({
```

Same for `OfferCard` - rename to a generic name?

File: src/components/OfferCard/OfferCard.jsx [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+                <Box fontSize="md" mb={4}>
+                  {props.description}
+                </Box>
+                {props.headerContentComponent}
+              </Box>
+              {hasOffers(props.offers) && (
+                <Box mt={[2, null, 'auto']}>
+                  <Label aria-label={props.offerLabel}>
+                    {props.offerLabel}
+                  </Label>
```
[the remark this answers is not in the document]
Comment (jacobdolsen) [reply src/components/OfferCard/OfferCard.jsx:59]: @ajlu Happy to go with your suggestion here. I was more thinking that its a label for the content below; ie. for offers that show up, it is a label and/or header to provide context for the offers shown.

File: src/components/Tags/Tags.jsx [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+  margin: 0;
+  > li + li {
+    &:before {
+      content: '${(props) => props.separator}';
+    }
+  }
+`;
+
+const Tags = (props) => (
+  <Box ml={[2, null, 2]} py={[0, null, 0]}>
```
Comment (ajlu) [inline src/components/Tags/Tags.jsx:18]:
```suggestion
  <Box ml={2} py={0}>
```

**Review by ajlu — APPROVED** [review]
Great work! 🚀

**Comment by ajlu** [comment]
Sorry, just remembered one more thing, can you add type defs for the new components to support typescript projects? I assume hotels could be interested in using these components

Check out `src/components/index.d.ts`

**Pull request merged**